web/src/stream_edit.js

```diff
@@ -665,21 +666,36 @@ export function initialize() {
         return true;
     });
 
+    const stream_save_changes = ($save_button) => {
+        const $subsection_elem = $save_button.closest(".settings-subsection-parent");
+
+        const stream_id = $save_button.closest(".subscription_settings.show").data("stream-id");
+        const sub = sub_store.get(stream_id);
+        const data = settings_org.populate_data_for_request($subsection_elem, false, sub);
+
+        const url = "/json/streams/" + stream_id;
+        settings_org.save_organization_settings(data, $save_button, url);
+    };
+
     $("#streams_overlay_container").on(
         "click",
         ".subsection-header .subsection-changes-save button",
         (e) => {
             e.preventDefault();
             e.stopPropagation();
             const $save_button = $(e.currentTarget);
-            const $subsection_elem = $save_button.closest(".settings-subsection-parent");
-
-            const stream_id = $save_button.closest(".subscription_settings.show").data("stream-id");
-            const sub = sub_store.get(stream_id);
-            const data = settings_org.populate_data_for_request($subsection_elem, false, sub);
 
-            const url = "/json/streams/" + stream_id;
-            settings_org.save_organization_settings(data, $save_button, url);
+            dialog_widget.launch({
+                html_heading: $t_html({defaultMessage: "Confirm stream privacy change"}),
+                html_body: render_confirm_stream_privacy_change_modal,
+                id: "confirm_stream_privacy_change",
+                html_submit_button: $t_html({defaultMessage: "Confirm"}),
+                html_exit_button: $t_html({defaultMessage: "Cancel"}),
```

Comment:
You don't need to pass the exit button parameter, I think, as it is the default.

Reply:
You're right, I've fixed it, thanks!